engine/classes/ElggEntity.php

```diff
@@ -1797,22 +1797,28 @@ public function disable($reason = "", $recursive = true) {
 
 			$ia = elgg_set_ignore_access(true);
 
-			$subentities = new \ElggBatch('elgg_get_entities', [
+			$base_options = [
 				'wheres' => [
-					"e.container_guid = $guid OR e.owner_guid = $guid OR e.site_guid = $guid",
-					"e.guid != $guid"
+					"e.guid != $guid",
 				],
-				'limit' => 0,
-			]);
-			$subentities->setIncrementOffset(false);
+				'limit' => false,
+			];
 			
-			foreach ($subentities as $subentity) {
-				/* @var $subentity \ElggEntity */
-				if (!$subentity->isEnabled()) {
-					continue;
+			foreach (['owner_guid', 'container_guid', 'site_guid'] as $db_column) {
```

Comment:
it will never matter... entities will only appear once... it does not matter in which result set